src/Core/functions.php

```diff
@@ -292,13 +292,24 @@ function deprecationWarning(string $message, int $stackFrame = 1): void
             $frame = $trace[$stackFrame];
             $frame += ['file' => '[internal]', 'line' => '??'];
 
+            $relative = str_replace(DIRECTORY_SEPARATOR, '/', substr($frame['file'], strlen(ROOT) + 1));
+            $patterns = (array)Configure::read('Error.ignoredDeprecationPaths');
+            foreach ($patterns as $pattern) {
+                $pattern = str_replace(DIRECTORY_SEPARATOR, '/', $pattern);
+                if (fnmatch($pattern, $relative)) {
+                    return;
+                }
+            }
+
             $message = sprintf(
                 '%s - %s, line: %s' . "\n" .
                 ' You can disable deprecation warnings by setting `Error.errorLevel` to' .
-                ' `E_ALL & ~E_USER_DEPRECATED` in your config/app.php.',
+                ' `E_ALL & ~E_USER_DEPRECATED`, or add `%s` to ' .
+                ' `Error.ignoredDeprecationPaths` in your `config/app.php` to mute deprecations.',
```

Comment:
"You can disable **all** deprecation warnings by  ...., or add .... to mute deprecations **in this file only**."